x-pack/plugins/cloud_security_posture/server/tasks/task_state.test.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep } from 'lodash';
+import { stateSchemaByVersion } from './task_state';
+
+describe('finding stats task state', () => {
+  describe('v1', () => {
```

Comment:
so the idea is, for subsequent versions we test each version individually?

Reply:
That is correct, for example a v2 schema will test different permutations of a v1 schema being migrated to v2 and so on for v3. The v1 schema is the most important migration because we take an unknown/loose state object and force a schema on it.